### PR TITLE
Sprinkle H5_ATTR_FORMAT over printf(3)-like functions in tools and fix issues

### PR DESCRIPTION
**Pull request by gnuoyd**

### CODE REVIEW

**Comment by lrknox** [comment]
Checks passed in action https://github.com/HDFGroup/hdf5/actions/runs/1797059611.

**Review by lrknox — APPROVED** [review]

File: tools/src/h5dump/h5dump_xml.c [hunk 1]
```diff
@@ -4529,12 +4526,11 @@ xml_print_enum(hid_t type)
                 h5tools_str_append(&buffer, "%02x", value[i * dst_size + j]);
         }
         else if (H5T_SGN_NONE == H5Tget_sign(native)) {
-            h5tools_str_append(&buffer, "%" H5_PRINTF_LL_WIDTH "u",
-                               *((unsigned long long *)((void *)(value + i * dst_size))));
+            /* XXX haven't I fixed this code somewhere else, before? */
```
Comment (byrnHDF) [inline tools/src/h5dump/h5dump_xml.c]: XXX?

**Review by byrnHDF — APPROVED** [review]

File: tools/src/h5dump/h5dump_xml.c (hunk 1 above)
Comment (lrknox) [reply tools/src/h5dump/h5dump_xml.c]: @gnuoyd is this comment to be kept, or was it a temporary note?

**Comment by gnuoyd** [comment]
On Fri, Feb 18, 2022 at 11:35:21AM -0800, Larry Knox wrote:
> @lrknox commented on this pull request.
> 
> 
> 
> > @@ -4529,12 +4526,11 @@ xml_print_enum(hid_t type)
>                  h5tools_str_append(&buffer, "%02x", value[i * dst_size + j]);
>          }
>          else if (H5T_SGN_NONE == H5Tget_sign(native)) {
> -            h5tools_str_append(&buffer, "%" H5_PRINTF_LL_WIDTH "u",
> -                               *((unsigned long long *)((void *)(value + i * dst_size))));
> +            /* XXX haven't I fixed this code somewhere else, before? */
> 
> @gnuoyd is this comment to be kept, or was it a temporary note?

I fixed the issues and removed the comments.

Dave

**Comment by lrknox** [comment]
All checks passed in action https://github.com/HDFGroup/hdf5/actions/runs/1878179798

**Review by byrnHDF — APPROVED** [review]

**Pull request merged**